Source/NameServer/AddressBook.cpp

```diff
@@ -0,0 +1,1361 @@
+//==============================================================================
+#include "AddressBook.hpp"
+//#include "Debug.h"
+
+#include <iostream>
+
+//==============================================================================
+namespace AddressBookNS
+{
+#ifdef EXCEPTS
```

Comment:
I've not seen this pattern before! Seems a little odd because you'll need to define `X` and `Y` in your source, even if they aren't needed...

Reply:
This was mine when I was considering having throwable exceptions.
I meant to talk to you about your preference - The exceptions really did not work well with Borland (it insisted on stepping down the exception tree) hence the return code option.

Reply:
I prefer exceptions for what you're doing here, because chains of `break` statements can be considerably more difficult to follow, but I'm pretty ambivalent.

I'd rather you choose one or the other for "production" - otherwise this becomes a bit of a Frankenstein amalgam.